### PR TITLE
Replace Pipeline with Node serialization/deserialization

### PR DESCRIPTION
**Pull request by jmilloy**

### CODE REVIEW

**Comment by jmilloy** [comment]
@mpu-creare I still have a few items to finish on #287, but this is ready to review and merge at any point.

**Comment by jmilloy** [comment]
@dsullivan-creare Could you review the AWS updates here? It includes the recent aws-automation merge. Alternatively, can you get me up to speed (or point me to documentation) so that I can test these changes myself?

File: podpac/core/node.py [hunk 1]
```diff
@@ -504,9 +506,191 @@ def rem_cache(self, key, coordinates=None, mode=None):
         self.cache_ctrl.rem(self, key=key, coordinates=coordinates, mode=mode)
 
     # --------------------------------------------------------#
-    #  Class Methods
+    #  Class Methods (Deserialization)
     # --------------------------------------------------------#
 
+    @classmethod
+    def from_definition(cls, definition):
```
Comment (mpu-creare) [inline podpac/core/node.py]: I'm assuming this was mostly copied and pasted from the pipeline module? Or should I check this carefully?

Comment (jmilloy) [reply podpac/core/node.py]: Yes, this is copied from the pipeline module. Same for the tests.

File: podpac/core/managers/aws_lambda.py [hunk 2]
```diff
@@ -66,15 +65,16 @@ def _AWS_REGION_NAME_default(self):
 
     source = tl.Instance(Node, allow_none=False, help="Node to evaluate in a Lambda function.")
 
-    source_output = tl.Instance(Output, allow_none=False, help="Image output information.")
+    source_output_format = tl.Unicode(default_value="png", allow_none=False, help="Image output format.")
```
[the remark this answers is not in the document]
Comment (jmilloy) [reply podpac/core/managers/aws_lambda.py]: This applies in the aws handler:

https://github.com/creare-com/podpac/blob/b197e2acfef1c88c7ab110dc8633775433d7ad3f/dist/aws/handler.py#L132

I think we can change the default to pkl.

[the remark this answers is not in the document]
Comment (jmilloy) [reply podpac/core/managers/aws_lambda.py]: a872a78

[the remark this answers is not in the document]
Comment (mpu-creare) [reply podpac/core/managers/aws_lambda.py]: I just remembered that the .pkl format is a pain. `netcdf` should be better.

**Comment by mpu-creare** [comment]
After @dsullivan-creare approves, I think we can merge this in.

**Pull request merged**